backend/tracim_backend/lib/core/content.py

```diff
@@ -2100,3 +2102,78 @@ def get_newest_authored_content(self, user_id: int) -> Content:
             return query.one()
         except NoResultFound as exc:
             raise ContentNotFound("User {} did not author any content".format(user_id)) from exc
+
+    def get_all_user_favorites(
+        self,
+        user_id: int,
+        content_type: str = content_type_list.Any_SLUG,
+        order_by_properties: typing.Optional[
+            typing.List[typing.Union[str, QueryableAttribute]]
+        ] = None,
+        page_token: typing.Optional[str] = None,
+        count: typing.Optional[int] = None,
+    ) -> Page:
+        """
+        Return all user favorite content using some filters
+        :param content_type: filter by content_type slug
+        :param order_by_properties: filter by properties can be both string of
+        attribute or attribute of Model object from sqlalchemy(preferred way,
+        QueryableAttribute object)
+        :return: List of contents
+        """
+        query = (
+            self.get_all_query(
+                parent_ids=None, content_type_slug=content_type, workspace=None, label=None,
+            )
+            .join(FavoriteContent, Content.id == FavoriteContent.content_id)
+            .filter(FavoriteContent.user_id == user_id)
+        )
+        for _property in order_by_properties:
+            query = query.order_by(_property)
+        if count:
+            return get_page(query, per_page=count, page=page_token or False)
+        return Page(query.all())
+
+    def get_one_user_favorite_content(self, user_id: int, content_id: int):
+        try:
+            return (
+                self.get_all_query()
+                .join(FavoriteContent, Content.id == FavoriteContent.content_id)
+                .filter(
+                    FavoriteContent.user_id == user_id, FavoriteContent.content_id == content_id
+                )
+                .one()
+            )
+        except NoResultFound:
+            raise ContentNotFound(
+                "Content of id {content_id} was not found as user {user_id} favorite.".format(
+                    content_id=content_id, user_id=user_id
+                )
+            )
+
+    def set_favorite(self, content: Content, do_save: bool = True) -> FavoriteContent:
```

Comment:
```suggestion
    def add_favorite(self, content: Content, do_save: bool = True) -> FavoriteContent:
```
I'm also unsure of having an implicit (current) user here as this is not the case for the get methods.

Reply:
done

Reply:
for implicit user, i just followed the way method are currently done, but we do have effectively a specific case for this one, don't make much sense to me to get an item without the full id (which is both user_id and content_id, getting only by content_id and guessing with current user seems strange to be).